carapace-server/src/main/java/org/carapaceproxy/core/ProxyRequest.java

```diff
@@ -171,6 +174,27 @@ public String getRequestHostname() {
         return request.requestHeaders().get(HttpHeaderNames.HOST);
     }
 
+    public String getHostname() {
+        String requestHostname = getRequestHostname();
+        int colonIndex = requestHostname.indexOf(":");
+        if (colonIndex != -1) {
+            String portString = requestHostname.substring(colonIndex + 1);
+            try {
+                int port = Integer.parseInt(portString);
+                if (port > 0 && port <= 65535) {
+                    return requestHostname.substring(0, colonIndex);
+                }
+            } catch (NumberFormatException e) {
+                return "";
```

Comment:
Maybe in that case is it better to return badRequest?

Reply:
Empty hostname is already mapped to badRequest in `MapResult map(ProxyRequest request)` because it is an invalid hostname.